controller/linodemachine_controller_helpers.go

```diff
@@ -431,21 +429,42 @@ func getVPCInterfaceConfig(ctx context.Context, machineScope *scope.MachineScope
 }
 
 func linodeMachineSpecToInstanceCreateConfig(machineSpec infrav1alpha2.LinodeMachineSpec) *linodego.InstanceCreateOptions {
-	var buf bytes.Buffer
-	enc := gob.NewEncoder(&buf)
-	err := enc.Encode(machineSpec)
-	if err != nil {
-		return nil
-	}
-
-	var createConfig linodego.InstanceCreateOptions
-	dec := gob.NewDecoder(&buf)
-	err = dec.Decode(&createConfig)
-	if err != nil {
-		return nil
+	interfaces := make([]linodego.InstanceConfigInterfaceCreateOptions, len(machineSpec.Interfaces))
+	for idx, iface := range machineSpec.Interfaces {
+		creatOpt := linodego.InstanceConfigInterfaceCreateOptions{
+			IPAMAddress: iface.IPAMAddress,
+			Label:       iface.Label,
+			Purpose:     iface.Purpose,
+			Primary:     iface.Primary,
+			SubnetID:    iface.SubnetID,
+			IPRanges:    iface.IPRanges,
+		}
+		if iface.IPv4 != nil {
```

Comment:
Instead of using iface.IPv4, should we use `iface.Purpose == linodego.InterfacePurposeVPC` here?

Reply:
this should be mapping machineSpec.Interfaces directly to instanceCOnfigOptions, I thought the logic to handle any interface specific configuration is handled in `getVPCInterfaceConfig`

Reply:
Yeah, first we have linodeMachineSpecToInstanceCreateConfig() and then getVPCInterfaceConfig(). I wonder why we need to set VPC config here.